### PR TITLE
Enabling Debug level logs in tests

### PR DESCRIPTION
**Pull request by AnomalRoil**
This is enabling Debug level logs in our tests, in order to help us debug our CI issues and flaky tests. (Most notably the stalled `make coverage` issues we seem to only have on the CI)

It is also adding some extra history to our race traces and stripping the specific GH prefixes from them, to allow for an easier usage when importing the logs locally.

### CODE REVIEW

**Review by willscott — DISMISSED** [review]

**Review by nikkolasg — DISMISSED** [review]

**Comment by AnomalRoil** [comment]
I think I'll try to refactor this to integrate it with the notion of "debug logs" of GHA: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging

Because the debug logs are quite verbose and not necessary in general.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    env:
+      DRAND_TEST_LOGS: "${{ runner.debug == '1' && 'DEBUG' || 'INFO' }}"
```
Comment (CluEleSsUK) [inline .github/workflows/tests.yaml:14]: perhaps I'm being dense, but what is `runner.debug == '1'` referring to? I couldn't see anything in github action docs

Comment (CluEleSsUK) [reply .github/workflows/tests.yaml:14]: I'm assuming it's what's stopping you extracting the env to the root of the file so it's shared between everything

Comment (AnomalRoil) [reply .github/workflows/tests.yaml:14]: https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context

But it seems it's not available there yet. So I'll try with the `RUNNER_DEBUG` env variable I guess.

Comment (CluEleSsUK) [reply .github/workflows/tests.yaml:14]: damned half-baked github features :<

**Review by CluEleSsUK — APPROVED** [review]

**Pull request merged**